src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -365,9 +366,12 @@ module.exports = function<T, P, I, TI, C>(
     workInProgress.firstEffect = null;
     workInProgress.lastEffect = null;
 
-    if (workInProgress.progressedPriority === priorityLevel) {
+    if (workInProgress.progressedPriority === priorityLevel ||
+        priorityLevel === TaskPriority) {
```

Comment:
I don't think that this is correct. Let's say we had a very low-priority render that had started deleting one child but it never completed. Instead it did some other higher priority updates to it. 

Then suddenly something in a boundary fails (or get another Task priority update). Then the child would be remounted instead of reused.

I don't have a better solution atm but I'll think about it some more. I don't fully understand how the failure case happens now.

Would always force unmounting children in an error boundary help with this case?